x/incentives/keeper/distribute.go

```diff
@@ -20,22 +21,28 @@ func (k Keeper) getDistributedCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
 	return coins
 }
 
+// Returns coins that have not been distributed yet from the provided gauges
 func (k Keeper) getToDistributeCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
 	// TODO: Consider optimizing this in the future to only require one iteration over all gauges.
+	// TODO: Lets make issue for this, requires changing getGaugesFromIterator logic in gauge.go
```

Comment:
TODO: Lets make issue for this, requires changing getGaugesFromIterator logic in gauge.go